sync-core/src/main/java/com/cloudant/sync/query/QuerySqlTranslator.java

```diff
@@ -250,6 +266,31 @@ private static QueryNode translateQuery(Map<String, Object> query,
             }
         }
 
+        // A text clause by nature uses its own text index it is handled
+        // separately from other simple clauses.
```

Comment:
Can you add an example of what `textClause` looks like to make it easier to follow the below.